cogs/schedulebuddy.py

```diff
@@ -0,0 +1,256 @@
+            self.aliases = self.raw_schedules["objects"]["aliases"]
+            return True
+
+    async def timeout(self, seconds = 60):
+        await asyncio.sleep(seconds)
+        await self.stop()
+
+    async def stop(self):
+        self.__bot.remove_listener(self.on_reaction_add)
+        await self.response_embed.clear_reactions()
+        print("Stopped Schedule Session")
+
+    def reset_timer(self):
+        if self.__timer:
+            if not self.__timer.cancelled():
+                self.__timer.cancel()
+        self.__timer = self.__bot.loop.create_task(self.timeout())
+
+    def build_embed(self, display_name, schedule) -> Embed:
+        embed = Embed(description=f'Listing **{self.current_page+1}**\
+            of **{len(self.pages)}** pages', color=0xb3edbd)
+        embed.set_author(name=display_name)
+        if self.aliases:
+            embed.set_footer(text="View class aliases at schedubuddy.com")
+        
+        image = draw_schedule(schedule)
+        bufferedio = BytesIO()
+        image.save(bufferedio, format="PNG")
+        bufferedio.seek(0)
+        file = discord.File(bufferedio, filename="image.png")
+        embed.set_image(url="attachment://image.png")
+        return embed, file
+
+
+    async def send_response(self, intr: discord.Interaction):
+        if not self.pages:
+            intr.response.send_message("No schedule found!")
+            return False
+        embed, file = self.build_embed(intr.user.display_name, self.pages[0])
+        channel = self.__bot.get_channel(intr.channel_id)
+        self.response_embed = await channel.send(embed=embed, file=file)
+        await self.response_embed.add_reaction(LEFT_EMOJI)
+        await self.response_embed.add_reaction(RIGHT_EMOJI)
+        return True
+    
+    async def on_reaction_add(self, reaction: discord.Reaction, user: discord.User):
+        if (reaction.message.id != self.response_embed.id) or \
+            (user.id != self.author.id):
+                return
+
+        emoji = str(reaction.emoji)
+        if emoji != LEFT_EMOJI and emoji != RIGHT_EMOJI:
+            return
+
+        if emoji == LEFT_EMOJI:
+            self.current_page -= 1
+        else:
+            self.current_page += 1
+
+        self.reset_timer()
+        self.current_page = self.current_page % len(self.pages)
+        embed, file  = self.build_embed(user.display_name, self.pages[self.current_page])
+        await self.response_embed.edit(embed=embed, attachments=[file])
+        await self.response_embed.remove_reaction(reaction, user)
+        print("Tried to update the schedule embed")
+
+
+class ScheduleBuddy(commands.GroupCog, name="schedubuddy"):
+    def __init__(self, bot):
+        self.http = None
+        self.__bot = bot
+
+    async def cog_load(self):
+        await super().cog_load()
+        print("ScheduleBuddy Cog loaded.")
+
+    async def init_http_client(self):
+        self.http = ClientSession()
```

Comment:
this doesn't need to be async does it? we can put in init which is clearer if it doesn't

Reply:
The `ClientSession` must be created in an async function. Creating it in a non-async function is deprecated.